File: desafio-04/kyldery/go/xadrez.go
package main

import (
	"fmt"
)

func main() {
	tabuleiro := [8][8]int{
		{4, 3, 2, 5, 6, 2, 3, 4},
		{1, 1, 1, 0, 0, 1, 1, 1},
		{0, 0, 0, 1, 1, 0, 0, 0},
		{0, 0, 0, 0, 0, 0, 0, 0},
		{0, 0, 0, 0, 0, 0, 0, 0},
		{0, 0, 0, 1, 1, 0, 0, 0},
		{1, 1, 1, 0, 0, 1, 1, 1},
		{4, 3, 2, 5, 6, 2, 3, 4},
	}
	
	nomeDasPecas := [6]string{
		"Peão",
		"Bispo",
		"Cavalo",
		"Torre",
		"Rainha",
		"Rei",
	}
	
	var totalDeCadaPeca [7]int
	
	for _, fileira := range tabuleiro {
		for _, casa := range fileira {
			totalDeCadaPeca[casa] += 1
		}
	}
	
	for i, total := range totalDeCadaPeca[1:] {
		fmt.Printf("%s: %d peça(s)\n", nomeDasPecas[i], total)
	}
}

File: desafio-04/g4br13lr0ch4/python/xadrez.py
"""
    Resolucao Desafio 4
    por: Gabriel Rocha
    github: G4BR13LR0CH4

"""
tabuleiro = []
peca = {"Peao" : 1, "Bispo" : 2, "Cavalo" : 3, "Torre": 4, "Rainha" : 5, "Rei" : 6}

def criacao():
    """Posiciona as pecas do tabuleiro"""
    for x in range(64):
        print("Casa: " + str(x))
        arm_peca = int(input("Digite o id da peca: "))
        tabuleiro.append(arm_peca)

def cont(flag, clone):
    """Faz a contagem das pecas"""
    while flag in clone:
        clone.remove(flag)
    return len(clone)

criacao()
for key in peca:
    aux = cont(peca[key], tabuleiro[:])
    peca[key] = 64 - int(aux)

for key in peca:
    print(key + ": " + str(peca[key]) + " peças")
